### PR TITLE
Move content of the Python pkg to subdirectory python/kvikio/

### PR DESCRIPTION
**Pull request by hcho3**
Part of #369

### CODE REVIEW

File: python/kvikio/pyproject.toml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2021-2022, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
```
Comment (wence-) [inline python/kvikio/pyproject.toml:1]: *nit*: I guess the automated copyright checker thinks this is a new file with no history? But I think this should be `2021-2024`?

**Review by wence- — APPROVED** [review]
Tiny niggle, approving python changes.

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
I think the devcontainer failures will require a change to https://github.com/rapidsai/devcontainers/blob/bff06939d44479ae1f63c3f128881d6d533de99c/features/src/rapids-build-utils/opt/rapids-build-utils/manifest.yaml#L48

This change would be similar to the changes in https://github.com/rapidsai/devcontainers/pull/263 when we moved `rmm`'s package directory from `python` to `python/rmm`.

**Comment by hcho3** [comment]
@bdice Let's merge this PR first. I would like to proceed as follows:

1. Merge this PR.
2. Submit a smaller PR to `devcontainer` to account for the directory change (`python` -> `python/kvikio`).
3. Merge #369 to add `libkvikio` and create C++ wheels
4. Submit a separate PR to  `devcontainer` to add `libkvikio`.

**Comment by bdice** [comment]
@hcho3 Sounds fine to me. You'll probably need to get admin-merges to work around the circular dependency between this PR and the one you plan to file to `rapidsai/devcontainers`.

**Comment by raydouglass** [comment]
Admin merging due to circular dependency

**Pull request merged**